Fixing semaphore checking for blackhole

jvegaTT opened the pull request:
### Problem description
Blackhole was having hangs when synchronizing devices after CCL OP. Root cause was that the OPs were not invalidating the L1 cache correctly when they were manually performing the semaphore checks. We have moved the manual checks to the common APIs which perform the correct invalidation.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes https://github.com/tenstorrent/tt-metal/actions/runs/16149658998
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable) https://github.com/tenstorrent/tt-metal/actions/runs/16150065976
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

jvegaTT committed:
Removed failing test

jvegaTT committed:
Removing offending tests

jvegaTT committed:
Removing offending tests

punithsekar committed:
Fix yolov6l CI failure (#24638)

### Problem description
(Single-card) Frequent model and ttnn tests CI test is failing due to
Yolov6l model,
https://github.com/tenstorrent/tt-metal/actions/runs/16047427538/job/45295642873#step:8:5137.

### What's changed
Fixed the issues on yolov6l.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [(Single-card) Frequent model and ttnn
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/fast-dispatch-full-regressions-and-models.yaml)
CI passes

SeanNijjar committed:
Enable Erisc IRAM by default when persistent fabric is launched for Wormhole (#20523)

Note that this PR was originally intended to enable erisc IRAM support
by default for all kernels. However, that was not possible due to
performance regressions for some legacy CCLs where the overhead to copy
to IRAM was an overall net loss in performance when compared to the
instruction fetch speedups.

Therefore global default enablement is deferred until legacy CCLs are
ported to run on fabric. For this PR, we instead only enable the IRAM
feature on Erisc when building fabric.

The overall flow for deciding if Erisc IRAM is enabled is then as
follows:
- if user specifies `TT_METAL_ENABLE_ERISC_IRAM` flag, override all
other behaviour with user choice
- else if/when fabric instantiated at device init time, enable IRAM
  - disable when torn down
- else Erisc iram disabled

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI: https://github.com/tenstorrent/tt-metal/actions/runs/16078141887
- [x] BH PC:
https://github.com/tenstorrent/tt-metal/actions/runs/16081848208
- [x] T3K unit:
https://github.com/tenstorrent/tt-metal/actions/runs/16078143816
- [x] T3K demo, frequent, nightly, model perf:
https://github.com/tenstorrent/tt-metal/actions/runs/16081850565
- demo failures same as main:
https://github.com/tenstorrent/tt-metal/actions/runs/16062787249
- frequent:
https://github.com/tenstorrent/tt-metal/actions/runs/16133050434 (after
testlist fixes)
- [x] TG:
https://github.com/tenstorrent/tt-metal/actions/runs/15982260830
- [x] TG Stress:
https://github.com/tenstorrent/tt-metal/actions/runs/15914430447
- [x] metal ubench:
https://github.com/tenstorrent/tt-metal/actions/runs/15916997396
  - same fails as main 
- [x] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI: https://github.com/tenstorrent/tt-metal/actions/runs/14866679747
- [x] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI: https://github.com/tenstorrent/tt-metal/actions/runs/14866681192
- [x] **(For models and ops writers)** Full [new models
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml)
CI: https://github.com/tenstorrent/tt-metal/actions/runs/14866686623

Closes #20340

jvegaTT committed:
Moved to safe form of semaphore checking

tt-rkim committed:
#0: [skip ci] Regress functional unet perf with trace from 2400 fps to 2300 fps (#24719)

### Ticket

N/A

### Problem description

This unet test is hella flaky.

### What's changed

Decrease threshold to something manageable.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI with demo tests passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml)
CI passes (if applicable) See [recommended dev
flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

Co-authored-by: Raymond Kim <raymond.kim243@outlook.com>

jvegaTT committed:
Merge branch 'main' into jvega/blackhole_ccl_fix

jvegaTT committed:
Small Oops

ntarafdar commented:
not sure if we have any BH tests in APC. @SeanNijjar can you comment on which pipeline those BH CCL tests are in?
@abhullar-tt likewise for BH Metal Ethernet tests.

abhullar-tt commented:
The BH post commit drop down has an option to run on LLM box which will include eth tests

cglagovichTT committed:
Minimal line reduce scatter (#24188)

### Problem description
Minimal ring-ReduceScatter needed to be extended to line topology in
order to support 4U galaxy and 6U submeshes.

### What's changed
- add support for minimal line RS
- use scatter-write to improve fabric bandwidth utilization
- move tests to nightly/t3k and nightly/tg, rename and clean up tests  

The implementation is 1 core per direction per link, line RS with some
synchronization methods to avoid race conditions between
forward/backward workers.
<img width="699" alt="image"
src="https://github.com/user-attachments/assets/ba10ac37-3f90-40ef-a84a-42209baaf6bc"
/>


### Checklist
- [x] New/Existing tests provide coverage for changes
- [x] TG Nightly
https://github.com/tenstorrent/tt-metal/actions/runs/15900850494
- [x] T3K Nightly
https://github.com/tenstorrent/tt-metal/actions/runs/15901005130

---------

Co-authored-by: jonathansuTT <jonathansu@tenstorrent.com>

sjameelTT committed:
Add 6U tests, a new TG perf pipeline and perf tests for all_to_all_dispatch (#24641)

### Ticket
#24669

### Problem description
Re-attempt with a new folder that doesn't run on post-commit by default.

previous attempt: https://github.com/tenstorrent/tt-metal/pull/24554
### What's changed

- New Perf pipeline for TG CCLs
- New 6U-specific tests

The benefit for a new folder/pipeline is that it doesn't add time to
post-commit, which runs all the CCLs and skips them when a mesh device
opens.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/16120573038/job/45485630853
- [x] [T3K Frequent]
https://github.com/tenstorrent/tt-metal/actions/runs/16082887181/job/45390244169
- [x] [TG Frequent]
(https://github.com/tenstorrent/tt-metal/actions/runs/16082887181/job/45390244169)

---------

Co-authored-by: Raymond Kim <109366641+tt-rkim@users.noreply.github.com>

tt-aho committed:
#24668: Fix call to memcpy to not read extra bytes (#24674)

### Ticket
https://github.com/tenstorrent/tt-metal/issues/24668

### Problem description
We sometimes issued memcpy calls that were larger than the src buffer,
as we were using the size of dst.

### What's changed
Copy only up to the src buffer size.

### Checklist
- [x] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/16132637718
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI with demo tests passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml)
CI passes (if applicable) See [recommended dev
flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [x] New/Existing tests provide coverage for changes

tt-rkim committed:
#0: Do not check compile time anymore for e2e perf models because we couldn't give less of a shit (#24601)

### Ticket

N/A

### Problem description

All infra team does is bump thresholds or skip these tests when compile
time regress.

And there has been a lack of meaningful intention or action from Model
writers.

### What's changed

Let's stop checking something and causing CI churn when there's no need
to.

### Checklist
- [ ] [All post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml)
CI passes
- [ ] [Blackhole Post
commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml)
CI with demo tests passes (if applicable)
- [ ] [Model
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml)
CI passes (if applicable)
- [ ] [Device performance
regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml)
CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo
tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml)
CI passes (if applicable) See [recommended dev
flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] New/Existing tests provide coverage for changes

Co-authored-by: Raymond Kim <raymond.kim243@outlook.com>

jvegaTT committed:
Fixed command processor all gather

jvegaTT committed:
Small oops

jvegaTT committed:
Forgot to re-cast semaphore set

jvegaTT committed:
Removed failing test

jvegaTT commented:
Appologies, I messed up during a rebase, closing this and re-opening
